src/huggingface_hub/file_download.py

```diff
@@ -229,6 +234,7 @@ def _raise_if_offline_mode_is_enabled(msg: Optional[str] = None):
 
 
 def _request_with_retry(
```

Comment:
Since this is private API, I don't think it needs to follow our deprecation policy here. Private API can change at any time w/o any guarantees for the user.